src/components/RightPane/RightPaneStore.ts

```diff
@@ -29,13 +29,25 @@ class RightPaneStore extends EventEmitter {
     private activeTab: number;
     private doDisplaySearch: boolean;
     private openSpotAlertPopoverActive: boolean;
+    private urlCourseCodeValue: string;
+    private urlTermValue: string;
+    private urlGEValue: string;
+    private urlCourseNumValue: string;
+    private urlDeptLabel: string;
+    private urlDeptValue: string;
     constructor() {
         super();
         this.setMaxListeners(15);
         this.formData = { ...defaultFormValues }; // creates shallow copy
         this.activeTab = 0;
         this.doDisplaySearch = true;
         this.openSpotAlertPopoverActive = false;
+        this.urlCourseCodeValue = String(new URLSearchParams(window.location.search).get('courseCode'));
+        this.urlTermValue = String(new URLSearchParams(window.location.search).get('term'));
+        this.urlGEValue = String(new URLSearchParams(window.location.search).get('GE'));
+        this.urlCourseNumValue = String(new URLSearchParams(window.location.search).get('courseNumber'));
+        this.urlDeptLabel = String(new URLSearchParams(window.location.search).get('deptLabel'));
+        this.urlDeptValue = String(new URLSearchParams(window.location.search).get('deptValue'));
```

Comment:
```suggestion
        this.urlCourseCodeValue = new URLSearchParams(window.location.search).get('courseCode').toString();
        this.urlTermValue = new URLSearchParams(window.location.search).get('term').toString();
        this.urlGEValue = new URLSearchParams(window.location.search).get('GE').toString();
        this.urlCourseNumValue = new URLSearchParams(window.location.search).get('courseNumber').toString();
        this.urlDeptLabel = new URLSearchParams(window.location.search).get('deptLabel').toString();
        this.urlDeptValue = new URLSearchParams(window.location.search).get('deptValue').toString();
```
I guess this is a preference; I rarely use the primitive constructors

Reply:
I think the `String()` constructor is better here. Because the `String()` function will call the `toString()` method of the object passed at the end. And also the `toString()` method cannot handle `null` or `undefined`, which will occur a lot in this scenario.

Reply:
Is this the reason why `"null"` can show up as form values? Hmm, what if we did something like this? I think all the results should either be string or nullish, and or-ing nullish results with an empty string will prevent "null" from ever being stringified. This could remove the need to check for `"null"` strings entirely?

```suggestion
        const search = new URLSearchParams(window.location.search)
        this.urlCourseCodeValue = search.get('courseCode') || '';
        this.urlTermValue = search.get('term') || '';
        this.urlGEValue = search.get('GE') || '';
        this.urlCourseNumValue = search.get('courseNumber') || '';
        this.urlDeptLabel = search.get('deptLabel') || '';
        this.urlDeptValue = search.get('deptValue') || '';
```